File: plugin/testData/shared/basics/Extensions.shared.kt
package com.test

typealias MyInteger = Int

infix fun Int.doubleAdd(other: Int): Int {
    return this + other + other
}

val Int.half: Int get() = this / 2

val MyInteger.double: Int get() = this * 2

operator fun String.div(other: String): String {
    return this + "/" + other
}

fun main(){
    println(3.toInt() doubleAdd 5)
    println(18.toInt().half)
    println(18.toInt().double)
    println("a" / "b")
}

File: plugin/testDataSwift/shared/basics/Extensions.shared.swift
//Package: com.test
//Converted using Khrysalis2

import Foundation



extension Int32 {
    public func doubleAdd(other: Int32) -> Int32 {
        return self + other + other
    }
    public func doubleAdd(_ other: Int32) -> Int32 {
        return doubleAdd(other: other)
    }
}
 
 

extension Int32 {
    public var half: Int32 {
        get {
            return self / 2
        }
    }
}
 
 

func /(self: String, other: String) -> String { return self.div(other: other) }
extension String {
    public func div(other: String) -> String {
        return self + "/" + other
    }
    public func div(_ other: String) -> String {
        return div(other: other)
    }
}
 
 

public func main() -> Void {
    print(3.toInt().doubleAdd(5))
    print(18.toInt().half)
    print("a" / "b")
}
 
